### PR TITLE
🥥 Handle proposal select type step

### LINKED ISSUE #834: Proposal type selection
@[figma](https://www.figma.com/file/pgNl83wgAub186krVkdiuo/Proposals?node-id=6%3A13)

Display a step where usere selects proposal type:

- Display a general proposal stepper
- Display proposal base parameters (deciding period, gracing limit, etc)
- Display proposal list

### LINKED ISSUE #885: Display Proposal Constants

### PR DESCRIPTION
**Pull request by DevNikita16**
Closes #834
Closes #885

### CODE REVIEW

File: packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx [hunk 1]
```diff
@@ -47,27 +65,66 @@ export const AddNewProposalModal = () => {
...
+    return <WarningModal onNext={() => send('NEXT')} />
+  }
+
+  if (state.matches('error')) {
+    return <FailureModal onClose={hideModal}>There was a problem while creating proposal.</FailureModal>
+  }
+
+  function selectType(type: ProposalDetails) {
+    setValid(true)
+    setType(type)
```
Comment (jodator) [inline packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx:96]: I think that we can persist the `type` in the machine's `context`. Adding typings to the machine will allow one to reliably access it in the later steps.

**Review by jodator — CHANGES_REQUESTED** [review]
It works nice, but the changes are needed:

1. The story for the select component
2. Enable only supported proposal types
3. Persisting state in the machine's `context` and providing better typings

File: packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx (hunk 1 above)
Comment (jodator) [reply packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx:96]: Example on the visualizer: https://xstate.js.org/viz/?gist=559858016b9fb97e46c8982617d267d1

File: packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  grid-template-columns: 1fr;
+  grid-template-rows: 1fr;
+  align-items: center;
+  width: 100%;
+  min-height: 100px;
+  padding: 8px 16px;
+  cursor: pointer;
+`
+
+export const TypeListItem = styled(ListItem)<{ active: boolean }>`
```
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx:56]: Where I can find list of enabled proposals?

[the remark this answers is not in the document]
Comment (jodator) [reply packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx:56]: There's none yet, so we need to code that as well. For now, let's assume that the proposal you plan to work on next will be enabled and the rest will be disabled.

We'll enable supported proposals one by one.

File: packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx (hunk 1 above)
Comment (DevNikita16) [reply packages/ui/src/proposals/modals/AddNewProposal/AddNewProposalModal.tsx]: ✅

File: packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx]: ✅

**Comment by DevNikita16** [comment]
I also added displaying constants in this PR

File: packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { proposalDescriptions } from '@/proposals/model/proposalDescriptions'
+import { ProposalDetails } from '@/proposals/types'
+
+interface TypeSelectionProps {
+  type: ProposalDetails | null
+  setType: (type: ProposalDetails) => void
+}
+
+export const TypeSelection = ({ type: chosenType, setType }: TypeSelectionProps) => {
+  return (
```
[the remark this answers is not in the document]
Comment (DevNikita16) [reply packages/ui/src/proposals/modals/AddNewProposal/components/TypeSelection.tsx]: ✅

**Review by jodator — APPROVED** [review]

**Pull request merged**